188 fix startstop markers should reset when choosing a new roll from the drop down perform mode only

cfitz opened the pull request:
This fixes the issue with start/stop markers persisting after using the roll selector drop-down.
Also adds some fixes for backwards rolls.

cfitz committed:
Fix start/stop markers persisting during roll changes. Also some bugs with backwards rolls

broadwell commented:
Thanks, this does indeed reset the start/stop makers when switching to a new roll, avoids the breaking errors with "backwards" (unrolling top to bottom) rolls, and generally streamlines the behavior of this feature (which turns out to be quite tricky to work out completely).
I'm still noticing some issues with the backwards-playing rolls, though, particularly that the markers and region highlight in the navigation bar still seem to be displaced from their correct positions, and don't always clear properly when the markers are deselected a new roll roll is chosen (see screenshot below).
If it would be helpful to merge this change as is and then open a new ticket and branch to handle these remaining edge cases, we can do that, or the final fixes can go in as additional commits here -- just let us know what you prefer.
![Screen Shot 2023-12-18 at 3 14 00 PM](https://github.com/sul-cidr/pianolatron/assets/7329112/1d7b2eea-98ea-4aa1-949f-fd5373837f51)

cfitz committed:
Merge branch 'develop' into 188-fix-startstop-markers-should-reset-when-choosing-a-new-roll-from-the-drop-down-perform-mode-only

cfitz committed:
Fixes for the nav bar overlays in the upsidedown

cfitz commented:
Added a few more fixes for the backwards rolls. 
There's some weirdness with OSD on these, but i think I sorted out most of the issues.
The fix is a bit hacky, but includes passing an empty onDraw function to the addOverlay arguments, which prevents OSD from trying to calculate the overlay position ( which can be off since not all the DOM elements have been loaded ). This seems to improve the accuracy/consistency of the overlay location a lot. 

https://deploy-preview-195--pianolatron-develop.netlify.app/perform/?druid=xy736dn5214&start=0.40&end=23.74

Changing rolls sometimes resulted in random markers being added in the navigator, but I think I fixed those. Will keep playing with it to see if any other cases pop up

cfitz committed:
Fix issue with tempomap on rolls without much tempo
